src/service/usage/mod.rs

```diff
@@ -209,23 +213,11 @@ async fn flush_triggers_usage() {
 
 async fn ingest_usages(curr_usages: Vec<UsageData>) {
     let mut groups: HashMap<GroupKey, AggregatedData> = HashMap::new();
+    let mut search_events = vec![];
     for usage_data in &curr_usages {
         // Skip aggregation for usage_data with event "Search"
         if usage_data.event == UsageEvent::Search {
-            let key = GroupKey {
-                stream_name: usage_data.stream_name.clone(),
-                org_id: usage_data.org_id.clone(),
-                stream_type: usage_data.stream_type,
-                day: usage_data.day,
-                hour: usage_data.hour,
-                event: usage_data.event,
-                email: usage_data.user_email.clone(),
-            };
-
-            groups.entry(key).or_insert_with(|| AggregatedData {
-                count: 1,
-                usage_data: usage_data.clone(),
-            });
+            search_events.push(usage_data.clone());
```

Comment:
Can we reduce the clone here?

Reply:
I guess no. `usage_data` is part of `curr_usages` which is used later as a backup in case the ingestion fails.

Reply:
got it